runtime/src/lib.rs

```diff
@@ -929,3 +929,98 @@ impl<A: AttributeWrite, const N: usize> Drop for LeaseBufWriter<A, N> {
         self.flush().ok();
     }
 }
+
+/// Abstraction over a buffer that reads bytes
+///
+/// This allows us to write code that's generic across both local slices and
+/// borrowed leases.
+pub trait BufReader<'a> {
+    /// Returns the size of the underlying buffer
+    fn len(&self) -> usize;
+
+    /// Reads a single byte, incrementing the internal position and returning
```

Comment:
Would it be more useful to return an iterator here than allow reading only one byte?

Reply:
Perhaps, but I'm matching the existing API (of `LeaseBufReader`), so don't want to mess with it too much.